internal/core/src/index/StringIndexSort.h

```diff
@@ -0,0 +1,431 @@
+#include <utility>
+#include <vector>
+#include <string>
+#include <map>
+#include <cstring>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <folly/small_vector.h>
+
+#include "index/StringIndex.h"
+#include "storage/MemFileManagerImpl.h"
+#include "storage/DiskFileManagerImpl.h"
+#include "storage/FileWriter.h"
+#include "common/File.h"
+
+namespace milvus::index {
+
+// Forward declaration
+class StringIndexSortImpl;
+
+// Main StringIndexSort class using pImpl pattern
+class StringIndexSort : public StringIndex {
+ public:
+    static constexpr uint32_t SERIALIZATION_VERSION = 1;
+    static constexpr uint64_t MAGIC_CODE =
+        0x5354524E47534F52;  // "STRNGSOR" in hex
+
+    explicit StringIndexSort(
+        const storage::FileManagerContext& file_manager_context =
+            storage::FileManagerContext());
+
+    virtual ~StringIndexSort();
+
+    int64_t
+    Count() override;
+
+    ScalarIndexType
+    GetIndexType() const override {
+        return ScalarIndexType::STLSORT;
+    }
+
+    const bool
+    HasRawData() const override {
+        return true;
+    }
+
+    void
+    Build(size_t n,
+          const std::string* values,
+          const bool* valid_data = nullptr) override;
+
+    void
+    Build(const Config& config = {}) override;
+
+    void
+    BuildWithFieldData(const std::vector<FieldDataPtr>& datas) override;
+
+    // See detailed format in StringIndexSortMemoryImpl::SerializeToBinary
+    BinarySet
+    Serialize(const Config& config) override;
+
+    IndexStatsPtr
+    Upload(const Config& config = {}) override;
+
+    void
+    Load(const BinarySet& index_binary, const Config& config = {}) override;
+
+    void
+    Load(milvus::tracer::TraceContext ctx, const Config& config = {}) override;
+
+    void
+    LoadWithoutAssemble(const BinarySet& binary_set,
+                        const Config& config) override;
+
+    // Query methods - delegated to impl
+    const TargetBitmap
+    In(size_t n, const std::string* values) override;
+
+    const TargetBitmap
+    NotIn(size_t n, const std::string* values) override;
+
+    const TargetBitmap
+    IsNull() override;
+
+    TargetBitmap
+    IsNotNull() override;
+
+    const TargetBitmap
+    Range(std::string value, OpType op) override;
+
+    const TargetBitmap
+    Range(std::string lower_bound_value,
+          bool lb_inclusive,
+          std::string upper_bound_value,
+          bool ub_inclusive) override;
+
+    const TargetBitmap
+    PrefixMatch(const std::string_view prefix) override;
+
+    std::optional<std::string>
+    Reverse_Lookup(size_t offset) const override;
+
+    int64_t
+    Size() override;
+
+ protected:
+    int64_t
+    CalculateTotalSize() const;
+
+    // Common fields
+    int64_t field_id_ = 0;
+    bool is_built_ = false;
+    Config config_;
+    std::shared_ptr<storage::MemFileManagerImpl> file_manager_;
+    size_t total_num_rows_{0};
+    TargetBitmap valid_bitset_;
+    std::vector<int32_t> idx_to_offsets_;
```

Comment:
this member is more `row_id_to_offset_` to me? the key(vector subscript) is the row id, the value is the offset of this row in `unique_values_`?